<a href="https://colab.research.google.com/github/sosarodrigox/grupo13_pp2/blob/main/ProyectoIntegrador_Grupo13.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Instituto Superior Politécnico Córdoba

## Tecnicatura Superior en Ciencia de Datos e Inteligencia Artificial

### Cátedra: Práctica Profesionalizante II

**Docente**: Facundo Oliva Cuneo

**Año**: 2024
  
**Trabajo Práctico Integrador**

## Grupo 13
### Integrantes:

- **Ingaramo, Ma. Eugenia**  
  [eugenia.ingaramo@gmail.com](mailto:eugenia.ingaramo@gmail.com)
- **Lonardi, Pablo**  
  [lonardipablo@gmail.com](mailto:lonardipablo@gmail.com)
- **Margheim, Carolina**  
  [caro08.m@gmail.com](mailto:caro08.m@gmail.com)
- **Sosa, Rodrigo**  
  [sosarodrigox@gmail.com](mailto:sosarodrigox@gmail.com)
- **Zenere, Mauricio**  
  [zeneremauricio@gmail.com](mailto:zeneremauricio@gmail.com)

In [4]:
import pandas as pd

# URL del archivo en formato 'Raw' en GitHub
url = "https://raw.githubusercontent.com/sosarodrigox/grupo13_pp2/main/data/data.csv"

# Cargar los datos en un DataFrame
data = pd.read_csv(url)

# Mostrar las primeras filas para confirmar la carga correcta
print(data.head())

          isbn13      isbn10           title subtitle  \
0  9780002005883  0002005883          Gilead      NaN   
1  9780002261982  0002261987    Spider's Web  A Novel   
2  9780006163831  0006163831    The One Tree      NaN   
3  9780006178736  0006178731  Rage of angels      NaN   
4  9780006280897  0006280897  The Four Loves      NaN   

                           authors                     categories  \
0               Marilynne Robinson                        Fiction   
1  Charles Osborne;Agatha Christie  Detective and mystery stories   
2             Stephen R. Donaldson               American fiction   
3                   Sidney Sheldon                        Fiction   
4              Clive Staples Lewis                 Christian life   

                                           thumbnail  \
0  http://books.google.com/books/content?id=KQZCP...   
1  http://books.google.com/books/content?id=gA5GP...   
2  http://books.google.com/books/content?id=OmQaw...   
3  http://books.go

In [5]:
# Configurar opciones de visualización
pd.set_option("display.max_columns", None)  # Todas las columnas
pd.set_option("display.max_rows", 10)  # 0 filas

# DataFrame
data

isbn13      isbn10                      title  \
0     9780002005883  0002005883                     Gilead   
1     9780002261982  0002261987               Spider's Web   
2     9780006163831  0006163831               The One Tree   
3     9780006178736  0006178731             Rage of angels   
4     9780006280897  0006280897             The Four Loves   
...             ...         ...                        ...   
6805  9788185300535  8185300534                  I Am that   
6806  9788185944609  8185944601       Secrets Of The Heart   
6807  9788445074879  8445074873             Fahrenheit 451   
6808  9789027712059  9027712050   The Berlin Phenomenology   
6809  9789042003408  9042003405  'I'm Telling You Stories'   

                                            subtitle  \
0                                                NaN   
1                                            A Novel   
2                                                NaN   
3                                                NaN   
4                                                NaN   
...                                              ...   
6805             Talks with Sri Nisargadatta Maharaj   
6806                                             NaN   
6807                                             NaN   
6808                                             NaN   
6809  Jeanette Winterson and the Politics of Reading   

                                           authors  \
0                               Marilynne Robinson   
1                  Charles Osborne;Agatha Christie   
2                             Stephen R. Donaldson   
3                                   Sidney Sheldon   
4                              Clive Staples Lewis   
...                                            ...   
6805  Sri Nisargadatta Maharaj;Sudhakar S. Dikshit   
6806                                 Khalil Gibran   
6807                                  Ray Bradbury   
6808                 Georg Wilhelm Friedrich Hegel   
6809                        Helena Grice;Tim Woods   

                         categories  \
0                           Fiction   
1     Detective and mystery stories   
2                  American fiction   
3                           Fiction   
4                    Christian life   
...                             ...   
6805                     Philosophy   
6806                      Mysticism   
6807                   Book burning   
6808                        History   
6809             Literary Criticism   

                                              thumbnail  \
0     http://books.google.com/books/content?id=KQZCP...   
1     http://books.google.com/books/content?id=gA5GP...   
2     http://books.google.com/books/content?id=OmQaw...   
3     http://books.google.com/books/content?id=FKo2T...   
4     http://books.google.com/books/content?id=XhQ5X...   
...                                                 ...   
6805  http://books.google.com/books/content?id=Fv_JP...   
6806  http://books.google.com/books/content?id=XcrVp...   
6807                                                NaN   
6808  http://books.google.com/books/content?id=Vy7Sk...   
6809  http://books.google.com/books/content?id=2lVyR...   

                                            description  published_year  \
0     A NOVEL THAT READERS and critics have been eag...          2004.0   
1     A new 'Christie for Christmas' -- a full-lengt...          2000.0   
2     Volume Two of Stephen Donaldson's acclaimed se...          1982.0   
3     A memorable, mesmerizing heroine Jennifer -- b...          1993.0   
4     Lewis' work on the nature of love divides love...          2002.0   
...                                                 ...             ...   
6805  This collection of the timeless teachings of o...          1999.0   
6806                                                NaN          1993.0   
6807                                                NaN          2004.0   
6808  Since the th

## Analisis de datos EDA:

In [6]:
# Obtener informacion del dataset recibido:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6810 entries, 0 to 6809
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   isbn13          6810 non-null   int64  
 1   isbn10          6810 non-null   object 
 2   title           6810 non-null   object 
 3   subtitle        2381 non-null   object 
 4   authors         6738 non-null   object 
 5   categories      6711 non-null   object 
 6   thumbnail       6481 non-null   object 
 7   description     6548 non-null   object 
 8   published_year  6804 non-null   float64
 9   average_rating  6767 non-null   float64
 10  num_pages       6767 non-null   float64
 11  ratings_count   6767 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 638.6+ KB


### Resumen del Dataset de Libros

El dataset analizado consta de **6,810 entradas**, cada una representa un libro único. Estos están descritos mediante **12 atributos**, que incluyen tanto identificadores numéricos como descriptivos. A continuación se detalla la estructura y las características principales del dataset:

### Identificadores de Libro
- `isbn13`: Cada libro tiene un identificador ISBN de 13 dígitos, proporcionado para todos los libros (6810 entradas no nulas).
- `isbn10`: Un identificador ISBN alternativo de 10 dígitos para cada libro, también completamente poblado.

### Títulos y Autores
- `title`: El título del libro, sin entradas nulas en el dataset.
- `subtitle`: Subtítulos proporcionados para 2,381 libros, sugiriendo que no todos los libros tienen un subtítulo explícito.
- `authors`: Nombres de los autores, presentes para 6,738 libros, indicando una pequeña proporción de entradas sin esta información. Esto podría representar un problema a futuro.

### Categorización y Descripción
- `categories`: Categorías o géneros bajo los cuales se clasifican los libros, con 6,711 libros categorizados. Hay casi 100 libros no categorizados lo cual podria generar conflictos a la hora de realizar recomendaciones.
- `description`: Descripciones textuales detalladas disponibles para 6,548 libros, proporcionando un contexto valioso sobre el contenido y el tema de cada libro y que podria utilizarce para las recomendaciones a clientes de la biblioteca.

### Aspectos Visuales y Editoriales
- `thumbnail`: URLs de imágenes de portada para 6,481 libros, lo que ayuda en la visualización y el marketing digital del libro. Se testeó valores al azar y se verificó que algunos links estaban caidos. Evaluar la solución para estos casos.
- `published_year`: Año de publicación registrado para 6,804 libros, ofreciendo una perspectiva sobre la temporalidad de las publicaciones.

### Calificaciones y Popularidad
- `average_rating`: La calificación promedio asignada por los usuarios, disponible para 6,767 libros, lo que refleja la recepción del libro por parte de los lectores.
- `num_pages`: Número de páginas, también disponible para 6,767 libros, indicando la extensión de los textos.
- `ratings_count`: El número de calificaciones que ha recibido cada libro, también notado para 6,767 libros, que puede ser utilizado para medir la popularidad o el engagement de los lectores.

**Uso del Dataset en Relación con los Objetivos del Proyecto**: Este conjunto de datos ofrece una base sólida para varios componentes esenciales del proyecto. *Es ideal para desarrollar un sistema automatizado de recomendación de libros*, tanto basado en contenido como en colaboración, alineándose con el objetivo de optimizar la experiencia del usuario y aumentar su satisfacción. Sin embargo, para respaldar completamente la iniciativa de mejorar la eficiencia en la gestión de recursos y optimizar la utilización de estos, sería beneficioso enriquecer el dataset con datos adicionales sobre las interacciones de los usuarios con la biblioteca, tales como:

 * Registros de préstamos.
 * Frecuencia de uso.
 * Retroalimentación directa de los usuarios.

Esta información adicional permitiría realizar análisis predictivos más precisos sobre tendencias de publicación y demanda de libros, facilitando una mejor alineación con las necesidades e intereses de los usuarios. Además, la integración y automatización de los registros de disponibilidad y estado de préstamo en el dataset podrían reducir significativamente la carga administrativa del personal, apoyando la eficiencia operativa y proporcionando una base de datos más completa sobre la cual tomar decisiones informadas para la gestión de colecciones y la programación de actividades de la biblioteca.

De todas formas, como se mencionó anteriormente el dataset es **suficiente** y presenta datos de **calidad** para una primera iteración en la construcción de un sistema de recomendación.